OMCompiler/Compiler/Util/UnorderedMap.mo

```diff
@@ -695,15 +721,16 @@ protected
     KeyEq eqfn = map.eqFn;
     list<Integer> bucket;
   algorithm
-    hash := hashfn(key, Vector.size(map.buckets));
-    bucket := Vector.get(map.buckets, hash + 1);
-
-    for i in bucket loop
-      if eqfn(key, Vector.getNoBounds(map.keys, i)) then
-        index := i;
-        break;
-      end if;
-    end for;
+    if Vector.size(map.buckets) > 0 then
```

Comment:
`hash` never gets a value here if there are no buckets. It might be better to just not allow creating such hashtables.

Reply:
thats true, for now i just pulled the hash computation out of the function

Reply:
I thought it was the modulo with 0 that the hash computation will do that was the issue, but maybe that's fine in MetaModelica.